x-pack/plugins/security_solution/public/timelines/components/side_panel/event_details/index.tsx

```diff
@@ -95,49 +98,38 @@ const EventDetailsPanelComponent: React.FC<EventDetailsPanelProps> = ({
     pagination,
   });
 
-  const hostRisk: HostRisk | null = data
-    ? {
-        loading: hostRiskLoading,
-        isModuleEnabled,
-        result: data,
-      }
-    : null;
+  const hostRisk: HostRisk | null = useMemo(() => {
+    return data
+      ? {
+          loading: hostRiskLoading,
+          isModuleEnabled,
+          result: data,
+        }
+      : null;
+  }, [data, hostRiskLoading, isModuleEnabled]);
 
-  if (!expandedEvent?.eventId) {
-    return null;
-  }
+  const { header, body, footer } = useMemo<{
```

Comment:
The following suggestion:

- separates the conditional logic for displaying the header from the conditional logic that displays the body
- eliminates the mutable variable `headerContent` and co-locates the `else` logic that was previously separate from the `isFlyoutView || isHostIsolationPanelOpen` condition
- eliminates the mutable variable `bodyContent`
- removes `footerContent` from the `useMemo`, because per the comment, it's the `Same footer for everyone`
 


```suggestion
  const header = useMemo(
    () =>
      isFlyoutView || isHostIsolationPanelOpen ? (
        <FlyoutHeader
          isHostIsolationPanelOpen={isHostIsolationPanelOpen}
          isAlert={isAlert}
          isolateAction={isolateAction}
          loading={loading}
          ruleName={ruleName}
          showAlertDetails={showAlertDetails}
          timestamp={timestamp}
        />
      ) : (
        <ExpandableEventTitle
          isAlert={isAlert}
          loading={loading}
          ruleName={ruleName}
          handleOnEventClosed={handleOnEventClosed}
        />
      ),
    [
      handleOnEventClosed,
      isAlert,
      isFlyoutView,
      isHostIsolationPanelOpen,
      isolateAction,
      loading,
      ruleName,
      showAlertDetails,
      timestamp,
    ]
  );

  const body = useMemo(() => {
    if (isFlyoutView) {
      return (
        <FlyoutBody
          alertId={alertId}
          browserFields={browserFields}
          detailsData={detailsData}
          event={expandedEvent}
          hostName={hostName}
          hostRisk={hostRisk}
          handleIsolationActionSuccess={handleIsolationActionSuccess}
          handleOnEventClosed={handleOnEventClosed}
          isAlert={isAlert}
          isDraggable={isDraggable}
          isolateAction={isolateAction}
          isIsolateActionSuccessBannerVisible={isIsolateActionSuccessBannerVisible}
          isHostIsolationPanelOpen={isHostIsolationPanelOpen}
          loading={loading}
          rawEventData={rawEventData}
          showAlertDetails={showAlertDetails}
          timelineId={timelineId}
          isReadOnly={isReadOnly}
        />
      );
    } else if (isHostIsolationPanelOpen) {
      return (
        <>
          {isIsolateActionSuccessBannerVisible && (
            <EndpointIsolateSuccess
              hostName={hostName}
              alertId={alertId}
              isolateAction={isolateAction}
            />
          )}
          <EuiFlyoutBody>
            <HostIsolationPanel
              details={detailsData}
              cancelCallback={showAlertDetails}
              successCallback={handleIsolationActionSuccess}
              isolateAction={isolateAction}
            />
          </EuiFlyoutBody>
        </>
      );
    } else {
      return (
        <>
          <EuiSpacer size="m" />
          <ExpandableEvent
            browserFields={browserFields}
            detailsData={detailsData}
            event={expandedEvent}
            isAlert={isAlert}
            isDraggable={isDraggable}
            loading={loading}
            rawEventData={rawEventData}
            timelineId={timelineId}
            timelineTabType={tabType}
            hostRisk={hostRisk}
            handleOnEventClosed={handleOnEventClosed}
          />
        </>
      );
    }
  }, [
    alertId,
    browserFields,
    detailsData,
    expandedEvent,
    handleIsolationActionSuccess,
    handleOnEventClosed,
    hostName,
    hostRisk,
    isAlert,
    isDraggable,
    isFlyoutView,
    isHostIsolationPanelOpen,
    isIsolateActionSuccessBannerVisible,
    isReadOnly,
    isolateAction,
    loading,
    rawEventData,
    showAlertDetails,
    tabType,
    timelineId,
  ]);

  if (!expandedEvent?.eventId) {
    return null;
  }

  return (
    <>
      {header}
      {body}
      <FlyoutFooter
        detailsData={detailsData}
        detailsEcsData={ecsData}
        expandedEvent={expandedEvent}
        refetchFlyoutData={refetchFlyoutData}
        handleOnEventClosed={handleOnEventClosed}
        isHostIsolationPanelOpen={isHostIsolationPanelOpen}
        isReadOnly={isReadOnly}
        loadingEventDetails={loading}
        onAddIsolationStatusClick={showHostIsolationPanel}
        timelineId={timelineId}
      />
    </>
  );
};
```